integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/complexType/TestComplexDataType.scala

```diff
@@ -111,19 +111,36 @@ class TestComplexDataType extends QueryTest with BeforeAndAfterAll {
       .addProperty(CarbonCommonConstants.CARBON_BAD_RECORDS_ACTION, "FAIL")
     sql("drop table if exists table1")
     sql("create table table1 (detail array<string>) stored by 'carbondata'")
-    sql("insert into table1 values('')")
+    //sql("insert into table1 values('')")
+    sql("insert into table1 values(array(''))")
     checkAnswer(sql("select detail[0] from table1"), Seq(Row("")))
     sql("drop table if exists table1")
     CarbonProperties.getInstance()
       .addProperty(CarbonCommonConstants.CARBON_BAD_RECORDS_ACTION, badRecordAction)
   }
 
+  test("test Projection PushDown for Struct - Int type when Array is Empty") {
+    CarbonProperties.getInstance()
+      .addProperty(CarbonCommonConstants.CARBON_BAD_RECORDS_ACTION, "FAIL")
+    sql("drop table if exists table1")
+    sql("create table table1 (detail array<int>) stored by 'carbondata'")
+    //sql("insert into table1 values ('\0011')")
+    sql("insert into table1 values(array())")
+    sql("select * from table1").show(false)
+    //checkAnswer(sql("select person.detail[0] from table1"), Seq(Row("")))
```

Comment:
@shivamasn : why this test case newly added ?

Reply:
@ajantha-bhat Removed